pkg/server/apis/v1alpha2/descriptors/stage.go

```diff
@@ -0,0 +1,219 @@
+	{
+		Path:        "/stages",
+		Description: "Stage APIs",
+		Definitions: []definition.Definition{
+			{
+				Method:      definition.Create,
+				Function:    handler.CreateStage,
+				Description: "Create stage",
+				Results:     definition.DataErrorResults("stage"),
+			},
+			{
+				Method:      definition.Get,
+				Function:    handler.ListStages,
+				Description: "List stages",
+				Parameters: []definition.Parameter{
+					{
+						Source:  definition.Query,
+						Name:    httputil.NamespaceQueryParameter,
+						Default: httputil.DefaultNamespace,
+					},
+				},
+				Results: definition.DataErrorResults("stages"),
+			},
+		},
+	},
+	{
+		Path:        "/stages/{stage}",
+		Description: "Stage APIs",
+		Definitions: []definition.Definition{
+			{
+				Method:      definition.Get,
+				Function:    handler.GetStage,
+				Description: "Get stage",
+				Parameters: []definition.Parameter{
+					{
+						Source: definition.Path,
+						Name:   httputil.StageNamePathParameterName,
+					},
+					{
+						Source:  definition.Query,
+						Name:    httputil.NamespaceQueryParameter,
+						Default: httputil.DefaultNamespace,
+					},
+				},
+				Results: definition.DataErrorResults("stage"),
+			},
+			{
+				Method:      definition.Update,
+				Function:    handler.UpdateStage,
+				Description: "Update stage",
+				Parameters: []definition.Parameter{
+					{
+						Source: definition.Path,
+						Name:   httputil.StageNamePathParameterName,
+					},
+				},
+				Results: definition.DataErrorResults("stage"),
+			},
+			{
+				Method:      definition.Delete,
+				Function:    handler.DeleteStage,
+				Description: "Delete stage",
+				Parameters: []definition.Parameter{
+					{
+						Source: definition.Path,
+						Name:   httputil.StageNamePathParameterName,
+					},
+					{
+						Source:  definition.Query,
+						Name:    httputil.NamespaceQueryParameter,
+						Default: httputil.DefaultNamespace,
+					},
+				},
+				Results: []definition.Result{definition.ErrorResult()},
+			},
+		},
+	},
+	{
+		Path: "/workflowruns/{workflowrun}/stages/{stage}/streamlogs",
+		Definitions: []definition.Definition{
+			{
+				Method:      definition.Get,
+				Function:    handler.ReceiveContainerLogStream,
+				Description: "Used for collecting stage logs",
+				Parameters: []definition.Parameter{
+					{
+						Source: definition.Path,
+						Name:   httputil.WorkflowRunNamePathParameterName,
+					},
+					{
+						Source: definition.Path,
+						Name:   httputil.StageNamePathParameterName,
+					},
+					{
+						Source: definition.Query,
+						Name:   httputil.ContainerNameQueryParameter,
+					},
+					{
+						Source:  definition.Query,
+						Name:    httputil.NamespaceQueryParameter,
+						Default: httputil.DefaultNamespace,
+					},
+				},
+				Results: []definition.Result{
+					{
+						Destination: definition.Error,
+					},
+				},
+			},
+		},
+	},
+	{
+		Path: "/workflowruns/{workflowrun}/stages/{stage}/logstream",
```

Comment:
I think this should be in `workflowruns` api group, not `stage`.

Reply:
If in `workflowruns` api group, we need to pass `stage` parameter.

Reply:
Keep the API path, just move it to `workflowrun.go`

Reply:
I will submit another pr to fix this after project API added.